zipline/finance/performance/period.py

```diff
@@ -143,14 +143,18 @@ def __init__(
         self.asset_finder = asset_finder
         self.data_frequency = data_frequency
 
+        # Start and end of the entire period
         self.period_open = period_open
         self.period_close = period_close
 
+        self.initialize(starting_cash=starting_cash,
```

Comment:
What's the difference between `__init__` and `initialize`?  What logic belongs in `__init__` and what belongs in `initialize`?  Splitting them has the significant cost of making it harder to find all the instance attributes of this class, so if there isn't an important reason to split them, I'd argue that we should do all our initialization in one place.